crates/fs/README.md

```diff
@@ -27,118 +27,129 @@
-        .mkdir(&["pictures".into(), "cats".into()], time, &store)
-        .await
-        .unwrap();
+use chrono::Utc;
+use rand::thread_rng;
 
-    // Get a sample CIDv1.
-    let cid = Cid::default();
+use std::rc::Rc;
 
-    // Add a file to /pictures/cats.
-    let OpResult { root_dir, .. } = root_dir
-        .write(
-            &["pictures".into(), "cats".into(), "tabby.png".into()],
-            cid,
-            time,
-            &store,
+#[async_std::main]
+async fn main() {
+    // Create a memory-based blockstore.
+    let store = &mut MemoryBlockStore::default();
+
+    // A random number generator the private filesystem can use.
+    let rng = &mut thread_rng();
+
+    // Create HAMT intermediate data structure.
+    let hamt = Rc::new(PrivateForest::new());
+
+    // Create a new private directory.
+    let dir = Rc::new(PrivateDirectory::new(
+        Namefilter::default(),
+        Utc::now(),
+        rng,
+    ));
+
+    // Add a file to /pictures/cats directory.
+    let PrivateOpResult { root_dir, hamt, .. } = dir
+        .mkdir(
+            &["pictures".into(), "cats".into()],
+            true,
+            Utc::now(),
+            hamt,
+            store,
+            rng,
         )
         .await
         .unwrap();
 
-    // Create and add a file to /pictures/dogs directory.
-    let OpResult { root_dir, .. } = root_dir
+    // Add a file to /pictures/dogs/billie.jpg file.
+    let PrivateOpResult { root_dir, hamt, .. } = root_dir
         .write(
             &["pictures".into(), "dogs".into(), "billie.jpeg".into()],
-            cid,
-            time,
-            &store,
+            true,
+            Utc::now(),
+            b"hello world".to_vec(),
+            hamt,
+            store,
+            rng,
         )
         .await
         .unwrap();
 
-    // Delete /pictures/cats directory.
-    let OpResult { root_dir, .. } = root_dir
-        .rm(&["pictures".into(), "cats".into()], &store)
-        .await
-        .unwrap();
-
     // List all files in /pictures directory.
-    let OpResult { result, .. } = root_dir
-        .ls(&["pictures".into()], &store)
+    let PrivateOpResult { result, .. } = root_dir
+        .ls(&["pictures".into()], true, hamt, store)
         .await
         .unwrap();
 
     println!("Files in /pictures: {:#?}", result);
 }
 ```
 
-## Building the Project
-
-- Build project
-
-  ```bash
-  cargo build --release
-  ```
-
-## Testing the Project
+Namefilters are currently how we represent the identity key of a node in the filesystem. They have nice properties, one of which is the ability to check if one node belongs to another. This is necessary in a filesystem where metadata like hierarchy needs to be hidden from observing agents. One notable caveat with namefilters is that they can only reliably store information of a file tree 47 levels deep or less so there is a plan to replace them with cryptographic accumlators in the near future.
```

Comment:
```suggestion
Namefilters are currently how we identify private node blocks in the filesystem. They have nice properties, one of which is the ability to check if one node belongs to another. This is necessary in a filesystem where metadata like hierarchy needs to be hidden from observing agents. One notable caveat with namefilters is that they can only reliably store information of a file tree 47 levels deep or less so there is a plan to replace them with other cryptographic accumlators in the near future.
```

I'd use "node blocks" instead of "node", because each namefilter only identifies a single block. The next revision will have a different identifier.
The identifier for nodes themselves would be `inumbers`. 

(The same paragraph may exist somewhere else and could maybe use the same suggestion?)